packages/mui-base/src/TextareaAutosize/TextareaAutoSize.tsx

```diff
@@ -0,0 +1,257 @@
+import * as React from 'react';
```

Comment:
There are too many changes in this file for git to treat it as a rename. It would be best to specify the new types in the TextareaAutosize.types.ts, to minimize the number of changes done here, so the history is preserved.
It would also make it easier to review the code.